thunder/core/module.py

```diff
@@ -141,6 +143,117 @@ def load_original_state_dict(self, state_dict):
                 else:
                     raise NotImplementedError(f"don't know how to handle {full_k}")
 
+    def state_dict(self, *, destination=None, prefix="", keep_vars=False):
+        """
+        Returns the state dict of the (transformed) Thunder module.
+
+        Args:
+            destination: if given, use this mutuable mapping as the dict container.
+            prefix: a prefix for the keys.
+            keep_vars: do not detach
+
+        Note that this is similar but rather more rudimentary than the original state_dict.
+        """
+        if destination is None:
+            destination = collections.OrderedDict()
+            destination._metadata = collections.OrderedDict()
```

Comment:
why OrderedDict? following https://github.com/pytorch/pytorch/blob/8624a571b4eecd11547867591d70992843265e97/torch/nn/modules/module.py#L2190-L2192?

Reply:
Yes, and because otherwise you cannot assign `._metadata` (I tried at first).